### PR TITLE
Top-K KL Divergence loss

### PR DESCRIPTION
**Pull request by AAnoosheh**
## What does this PR do?

**Type of change:** New feature

**Overview:** Writes a new KLDiv Logits loss which only uses top-k vocab values

## Usage

```python
# Add a code snippet demonstrating how to use this
```

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes/No 
- **Did you write any new necessary tests?**: Yes/No
- **Did you add or update any necessary documentation?**: Yes/No
- **Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?**: Yes/No 

## Summary by CodeRabbit

* **New Features**
  * Added Top-K logit filtering capability for knowledge distillation workflows, enabling selective focus on high-probability tokens.

* **Improvements**
  * Enhanced distributed tensor model-parallel operations with improved awareness for gradient computation and reduction.
  * Simplified legacy distributed operation constructs.

* **Tests**
  * Introduced comprehensive test coverage for Megatron-based distillation, validating both standard and Top-K filtering variants.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: modelopt/torch/distill/plugins/megatron.py [hunk 1]
```diff
@@ -310,81 +319,143 @@ def forward(self, predictions: Tensor, targets: Tensor) -> Tensor:
...
 
-            denom_student = torch.sum(torch.exp(output_student), dim=-1)
-            denom_student = all_reduce_autograd(
-                denom_student, group=parallel_state.get_tensor_model_parallel_group()
-            )
+            # Compute global softmax denominators
+            # We can't use standard all_reduce function here since the computation
+            # that follows it isn't identical across TP ranks.
+            denom_teacher = torch.sum(torch.exp(output_teacher), dim=-1, keepdim=True)
+            denom_teacher = dist_nn.functional.all_reduce(denom_teacher, group=tp_group)
```
Comment (kevalmorabia97) [inline modelopt/torch/distill/plugins/megatron.py:350]: Is `torch.distributed.nn.functional.all_reduce` same as `torch.distributed.all_reduce`?

Comment (AAnoosheh) [reply modelopt/torch/distill/plugins/megatron.py:350]: No, it's the premium edition which allows gradient backprop through it
